React/CSS-in-JS/Chakra.md

```diff
@@ -0,0 +1,498 @@
+
+> Under the hood `@media(min-width)` media query is used to ensure mobile-first
+
+Chakra UI default breakpoints:
+
+```ts
+export const breakpoints = {
+  sm: "30em",
+  md: "48em",
+  lg: "62em",
+  xl: "80em",
+};
+```
+
+Here's how to interpret this syntax:
+
+- `base`: From `0em` upwards
+- `md`: From `48em` upwards
+- `lg`: From `62em` upwards
+- `xl`: From `80em` upwards
+
+For responsive styles, array or object syntax can be used.
+
+```tsx
+// array syntax
+<Text fontSize={["24px", "40px", "56px"]}>
+  Lorem Ipsum is simply dummy text
+</Text>
+```
+
+```tsx
+// object syntax
+<Text fontSize={{ base: "24px", md: "40px", lg: "56px" }}>
+  Lorem Ipsum is simply dummy text
+</Text>
+```
+
+In case we need to skip cretan breakpoint, `null` is passed at that position in the array to avoid generating unnecessary CSS.
+
+Example:
+
+```jsx
+<Text fontSize={["24px", null, "56px"]}>Lorem Ipsum is simply dummy text</Text>
+```
+
+Array and Object syntax work for every style prop in the them specification, which means you can change most properties at a given breakpoint.
+
+To create custom breakpoints read [docs](https://chakra-ui.com/docs/features/responsive-styles#customizing-breakpoints).
+
+### Custom components style
+
+Chakra UI has a specific API for styling components. Most components have default or base styles `baseStyle`, styles for different sizes `sizes`, and styles for different visual variants `variants`.
+
+Components styles are defined in `theme/components/<component-name>.ts`
+Each component style will export these objects:
+
+- `baseStyles`
+- `sizes`
+- `variants`
+- `defaultProps`
+
+Base style are styles that all button types share.
+
+```tsx
+const baseStyle = {
+  lineHeight: "1.2",
+  borderRadius: "md",
+  _focus: {
+    boxShadow: "outline",
+  },
+  _disabled: {
+    opacity: 0.4,
+    cursor: "not-allowed",
+    boxShadow: "none",
+  },
+  _hover: {
+    cursor: "pointer",
+  },
+};
+```
+
+<!-- TODO change this example to use color schema -->
+
+Variant represents visual style.
+
+```tsx
+const variants = {
+  solid: {
+    bg: 'primary.100',,
+    color: 'white',
+    _hover: {
+      bg: 'primary.400',
+    },
+    _disabled: {
+      bg: 'neutral.200',
+    }
+  },
+	outline: {
+    bg: 'transparent',,
+    border: '1px solid',
+    borderColor: 'primary.100',
+    _hover: {
+      bg: 'primary.100',
+    }
+  },
+};
+```
+
+Sizes determent all the component sizes like `width`, `height`, `font-size` etc
+
+```tsx
+const sizes = {
+  sm: {
+    h: 8,
+    minW: 8,
+    fontSize: "sm",
+    px: 3,
+  },
+  xs: {
+    h: 6,
+    minW: 6,
+    fontSize: "xs",
+    px: 2,
+  },
+};
+```
+
+Naming used for sizes:
+
+- `lg`
+- `md`
+- `sm`
+- `xs`
+
+Default values are default values for `size` and `variant`
```

Comment:
This can be a bit confusing to read, maybe change it to this?
```suggestion
`defaultProps` are default values for `size` and `variant`
```

Similar comment for `Sizes determent all` as well, maybe change it to
```
`sizes` determine....
```